test/controllers/auth_controller_test.rb

```diff
@@ -19,15 +23,37 @@ class HealthCardsControllerTest < ActionDispatch::IntegrationTest
     get(auth_code_path,
         params: { client_id: Rails.application.config.client_id, redirect_uri: redirect_uri, state: state })
     assert_redirected_to "#{redirect_uri}?code=#{Rails.application.config.auth_code}&state=#{state}"
+    assert_no_cache_headers
   end
 
-  test 'token with incorrect code' do
-    post(auth_token_path, params: { code: 'code' })
-    assert_response :unauthorized
+  test 'token with incorrect parameter should return 400 invalid_request' do
+    post(auth_token_path)
+    assert_response :bad_request
+    assert_no_cache_headers
+    assert_equal('{"error":"invalid_request"}', @response.body)
+  end
+
+  test 'token with incorrect code should return 400 invalid_client' do
+    post(auth_token_path, params: { code: 'bad_code' })
+    assert_response :bad_request
+    assert_no_cache_headers
+    assert_equal('{"error":"invalid_client"}', @response.body)
   end
 
   test 'token with correct code' do
```

Comment:
Kind of nit-picky, but can we add what it "should" do to the title of this test since all of the other tests are formatted that way now.

Reply:
Fixed